### PR TITLE
resume/suspend autoscaling for a ASG

### PR DESCRIPTION
**Pull request by danhimalplanet**
add resuming and suspending autoscaling in a ASG functionaliy

I'm still working on making it work before and after a code deploy.

### CODE REVIEW

File: TODO.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+### TODO
+* Parallel-SSH integration (needs to be merged from another repo)
+* CodeDeploy support from S3
+* For all ASG in a CodeDeploy group, suspend all autoscaling processes before a CodeDeploy deploy ( if there are autoscaling rules in the ASG).  After deploy is done, resume autoscaling processes in the ASG.  I added suspend and resume, I wasn't able to successfully suspend before a CodeDeploy deploy.  Working on it.
```
Comment (revmischa) [inline TODO.md]: what's with the extra space after (

File: udo/asgroup.py [hunk 2]
```diff
@@ -362,3 +363,101 @@ def ip_addresses(self):
...
+        print("Autoscaling Status:"),
+        if self.suspend_status():
+            print(colored("SUSPENDED", 'yellow'))
+        else:
+            print(colored("ACTIVE", 'green'))
+
+        print('')
+        for asg_policy in asg_policies:
+            debug(asg_policy)
+            ScalingPolicies = []
```
Comment (revmischa) [inline udo/asgroup.py]: inconsistent CamelCase when the rest of udo is under_scores

File: udo/asgroup.py [hunk 3]
```diff
@@ -362,3 +363,101 @@ def ip_addresses(self):
...
+            if ScalingPolicies:
+                print(colored("unhandled policies: ")),
+                print(ScalingPolicies)
+            
+            print("")
+
+    def suspend(self):
+        debug("In asgroup.py suspend")
+        name = self.name()
+        asg_policies = self.conn.describe_policies( AutoScalingGroupName = name )['ScalingPolicies']
```
Comment (revmischa) [inline udo/asgroup.py:417]: should prob be refactored into a method

File: udo/main.py [hunk 4]
```diff
@@ -193,7 +203,10 @@ def get_deployment_group_name(self, group_arg):
 
     def version(self, *args):
         args = list(args)
-        print('3.0.2')
+        try:
+            print(pkg_resources.get_distribution('udo').version)
```
Comment (revmischa) [inline udo/main.py:207]: cool

File: TODO.md (hunk 1 above)
Comment (danhimalplanet) [inline TODO.md]: fixed

File: udo/asgroup.py (hunk 2 above)
Comment (danhimalplanet) [inline udo/asgroup.py]: fixed

File: udo/asgroup.py (hunk 3 above)
Comment (danhimalplanet) [inline udo/asgroup.py:417]: fixed

**Pull request merged**